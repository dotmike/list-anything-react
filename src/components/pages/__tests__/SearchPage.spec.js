import configureStore from "redux-mock-store";

import ConnectedSearchPage, { UnconnectedSearchPage } from "../SearchPage";

const testData = {};
testData.query = "test";
testData.search = `?query=${testData.query}`;
testData.newQuery = "foo";
testData.newSearch = `?query=${testData.newQuery}`;

const setup = (propOverrides = {}) => {
  const props = {
    history: {
      push: jest.fn()
    },
    location: {
      search: testData.search
    },
    results: [],
    searchRequest: jest.fn(() => Promise.resolve()),
    ...propOverrides
  };

  return {
    props,
    connectedWrapperShallow: wrapperShallow(ConnectedSearchPage, props),
    wrapperShallow: wrapperShallow(UnconnectedSearchPage, props)
  };
};

describe("ConnectedSearchPage", () => {
  const mockStore = configureStore();
  const state = {};
  it("renders correctly", () => {
    const { connectedWrapperShallow: wrapper } = setup({
      store: mockStore(state)
    });
    expect(wrapper()).toMatchSnapshot();
  });
});

describe("UnconnectedSearchPage", () => {
  it("renders correctly before `searchRequest` finishes", () => {
    const { wrapperShallow: wrapper, props } = setup();
    expect(wrapper()).toMatchSnapshot();
    expect(props.searchRequest).toHaveBeenCalledWith(testData.query);
  });

  it("renders correctly after `searchRequest` finishes successfully", done => {
    const { wrapperShallow: wrapper, props } = setup();

    wrapper();
    expect(props.searchRequest).toHaveBeenCalledWith(testData.query);

    setImmediate(() => {
      wrapper().update();
      expect(wrapper()).toMatchSnapshot();
      done();
    });
  });

  it("renders correctly after `searchRequest` finishes with failure", done => {
    const { wrapperShallow: wrapper, props } = setup({
      searchRequest: jest.fn(() => Promise.reject())
    });

    wrapper();
    expect(props.searchRequest).toHaveBeenCalledWith(testData.query);

    setImmediate(() => {
      wrapper().update();
      expect(wrapper()).toMatchSnapshot();
      done();
    });
  });

  it("calls `searchRequest` again after `location.search` changes", () => {
    const { wrapperShallow: wrapper, props } = setup({
      searchRequest: jest.fn(() => Promise.reject())
    });

    wrapper().setProps({
      location: { search: testData.newSearch }
    });

    expect(props.searchRequest).toHaveBeenCalledWith(testData.newQuery);
  });
});
