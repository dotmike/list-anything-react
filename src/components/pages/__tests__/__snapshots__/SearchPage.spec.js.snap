// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedSearchPage renders correctly 1`] = `
<SearchPage
  history={
    Object {
      "push": [Function],
    }
  }
  location={
    Object {
      "search": "?query=test",
    }
  }
  results={Array []}
  searchRequest={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
/>
`;

exports[`UnconnectedSearchPage renders correctly after \`searchRequest\` finishes successfully 1`] = `
<Segment
  basic={true}
  loading={false}
  style={
    Object {
      "margin": "10px auto",
      "maxWidth": "90%",
    }
  }
>
  <SearchResultTable
    onTableRowClick={[Function]}
    results={Array []}
  />
</Segment>
`;

exports[`UnconnectedSearchPage renders correctly after \`searchRequest\` finishes with failure 1`] = `
<Segment
  basic={true}
  loading={false}
  style={
    Object {
      "margin": "10px auto",
      "maxWidth": "90%",
    }
  }
>
  <ErrorMessage
    showHeader={true}
    text=""
  />
</Segment>
`;

exports[`UnconnectedSearchPage renders correctly before \`searchRequest\` finishes 1`] = `
<Segment
  basic={true}
  loading={true}
  style={
    Object {
      "margin": "10px auto",
      "maxWidth": "90%",
    }
  }
>
  <SearchResultTable
    onTableRowClick={[Function]}
    results={Array []}
  />
</Segment>
`;
